owncloud/owncloud.py

```diff
@@ -1016,6 +1016,28 @@ def search_users(self, user_name):
 
         raise HTTPResponseError(res)
 
+    def get_users(self):
```

Comment:
I think it's fine to let `search_users` accept an empty string.

So let's use your last solution, the one with the code from https://github.com/owncloud/pyocclient/pull/187#discussion_r103673476.

As a bonus, it would be good to discard the "?search=" bit if no search string is passed.